### PR TITLE
[14.0][MIG] mass_mailing_partner: Migration to 14.0

### PR DESCRIPTION
**Pull request by AreebSiddiqiIR**

### CODE REVIEW

**Review by moylop260 — APPROVED** [review]
👍

**Comment by moylop260** [comment]
@victoralmau @Tardo @jaredkipe 

Could you check it, please?

File: mass_mailing_partner/models/mailing_contact.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        record = self.new(vals)
+        if not record.partner_id:
+            record._set_partner()
+        record._onchange_partner_mass_mailing_partner()
+        new_vals = record._convert_to_write(record._cache)
+        new_vals.update(
+            subscription_list_ids=vals.get("subscription_list_ids", []),
+            list_ids=vals.get("list_ids", []),
+        )
+        return super(MailingContact, self).create(new_vals)
```
Comment (victoralmau) [inline mass_mailing_partner/models/mailing_contact.py]:
```suggestion
        return super().create(new_vals)
```

File: mass_mailing_partner/models/mailing_contact.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            record = self.new(new_vals)
+            if not record.partner_id:
+                record._set_partner()
+            record._onchange_partner_mass_mailing_partner()
+            new_vals = record._convert_to_write(record._cache)
+            new_vals.update(
+                subscription_list_ids=vals.get("subscription_list_ids", []),
+                list_ids=vals.get("list_ids", []),
+            )
+            super(MailingContact, contact).write(new_vals)
```
Comment (victoralmau) [inline mass_mailing_partner/models/mailing_contact.py:73]:
```suggestion
            super().write(new_vals)
```

File: mass_mailing_partner/models/mailing_trace.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    def partner_link(self):
+        for stat in self.filtered(lambda r: r.model and r.res_id):
+            partner_id = self.partner_id_from_obj(stat.model, stat.res_id)
+            if partner_id != stat.partner_id.id:
+                stat.partner_id = partner_id
+        return True
+
+    @api.model
+    def create(self, vals):
+        stat = super(MailingTrace, self).create(vals)
```
Comment (victoralmau) [inline mass_mailing_partner/models/mailing_trace.py]:
```suggestion
        stat = super().create(vals)
```

File: mass_mailing_partner/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        )
+        mapped_data = {
+            contact["partner_id"][0]: contact["partner_id_count"]
+            for contact in contact_data
+        }
+        for partner in self:
+            partner.mass_mailing_stats_count = mapped_data.get(partner.id, 0)
+
+    def write(self, vals):
+        res = super(ResPartner, self).write(vals)
```
Comment (victoralmau) [inline mass_mailing_partner/models/res_partner.py]:
```suggestion
        res = super().write(vals)
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code review OK. Only little changes.

File: mass_mailing_partner/models/mailing_contact.py (hunk 2 above)
Comment (AreebSiddiqiIR) [reply mass_mailing_partner/models/mailing_contact.py:73]: @victoralmau Thanks for your review.

This `super` here inside a for-loop and I don't think it should be changed.

**Comment by victoralmau** [comment]
Thanks @AreebSiddiqiIR but please, squash last commit to "Migration to 14.0"

**Comment by moylop260** [comment]
@AreebSiddiqiIR 

He is asking for doing the following process (see video):
 - [IMAGE ALT TEXT HERE](https://www.youtube.com/watch?v=apZtl15dHuo)

**Comment by moylop260** [comment]
The review was already fixed so merging
Thank you!

File: mass_mailing_partner/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_partner_mail_list_wizard,access_partner_mail_list_wizard,model_partner_mail_list_wizard,base.group_user,1,0,0,0
```
Comment (moylop260) [inline mass_mailing_partner/security/ir.model.access.csv]: @victoralmau 
Are you agree with this change in the last commit?

**Comment by moylop260** [comment]
@AreebSiddiqiIR 

Could you run `pre-commit run --all`, please?

And rebase the new commit, that pre-commit will generate, in the last one

File: mass_mailing_partner/security/ir.model.access.csv (hunk 5 above)
Comment (victoralmau) [inline mass_mailing_partner/security/ir.model.access.csv]:
```suggestion
access_partner_mail_list_wizard,access_partner_mail_list_wizard,model_partner_mail_list_wizard,base.group_user,1,1,1,0
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Review by victoralmau — APPROVED** [review]

**Comment by samannisar** [comment]
:+1:

**Comment by samannisar** [comment]
Approved

**Pull request merged**